File: tasks/translation/output/Qwen/Qwen2.5-32B-Instruct-AWQ/codenet/cot/Java/C++/s824230885_commented.cpp
#include <iostream>
#include <vector>
#include <string>
#include <map>
#include <sstream>
using namespace std;

namespace Main {
    class InputIterator {
        vector<string> inputLine;
        int index = 0, max;
        string read;
    public:
        InputIterator() {
            while (getline(cin, read)) {
                inputLine.push_back(read);
            }
            max = inputLine.size();
        }
        bool hasNext() { return (index < max); }
        string next() {
            if (hasNext()) {
                return inputLine[index++];
            } else {
                throw out_of_range("There is no more input");
            }
        }
    };
    map<int, string> CONVSTR;
    InputIterator ii;
    void flush() { cout.flush(); }
    void myout(const string& t) { cout << t << endl; }
    void myerr(const string& t) { cerr << "debug:" << t << endl; }
    string next() { return ii.next(); }
    bool hasNext() { return ii.hasNext(); }
    int nextInt() { return stoi(next()); }
    long long nextLong() { return stoll(next()); }
    double nextDouble() { return stod(next()); }
    vector<string> nextStrArray() { return myconv(next(), 8); }
    vector<string> nextCharArray() { return myconv(next(), 0); }
    vector<int> nextIntArray() {
        vector<string> input = nextStrArray();
        vector<int> ret(input.size());
        for (int i = 0; i < input.size(); i++) {
            ret[i] = stoi(input[i]);
        }
        return ret;
    }
    vector<long long> nextLongArray() {
        vector<string> input = nextStrArray();
        vector<long long> ret(input.size());
        for (int i = 0; i < input.size(); i++) {
            ret[i] = stoll(input[i]);
        }
        return ret;
    }
    string myconv(const string& list, int no) {
        string joinString = CONVSTR[no];
        if (list.find(joinString) != string::npos) {
            return list;
        } else {
            throw invalid_argument("Don't join");
        }
    }
    vector<string> myconv(const string& str, int no) {
        string splitString = CONVSTR[no];
        vector<string> ret;
        stringstream ss(str);
        string item;
        while (getline(ss, item, splitString[0])) {
            ret.push_back(item);
        }
        return ret;
    }
    void solve() {
        CONVSTR[8] = " ";
        CONVSTR[9] = "\n";
        CONVSTR[0] = "";
        int N = nextInt();
        vector<int> list = nextIntArray();
        int max = 0;
        int output = 0;
        for (int i = 0; i < N; i++) {
            if (max <= list[i]) {
                max = list[i];
                output++;
            }
        }
        myout(to_string(output));
    }
}

int main() {
    Main::solve();
    Main::flush();
    return 0;
} // End of Code

File: tasks/translation/output/Qwen/Qwen2.5-32B-Instruct-AWQ/codenet/cot/Java/Python/s824230885_commented.py
import sys
from typing import List

class InputIterator:
    def __init__(self):
        self.input_line = [line.strip() for line in sys.stdin.readlines()]
        self.index = 0
        self.max = len(self.input_line)

    def hasNext(self):
        return self.index < self.max

    def next(self):
        if self.hasNext():
            return self.input_line[self.index]
        else:
            raise IndexError("There is no more input")

ii = InputIterator()
out = sys.stdout.write
flush = sys.stdout.flush
myout = lambda t: out(f"{t}\n")
myerr = lambda t: sys.stderr.write(f"debug: {t}\n")
next = ii.next
hasNext = ii.hasNext
nextInt = lambda: int(next())
nextLong = lambda: int(next())
nextDouble = lambda: float(next())
nextStrArray = lambda: next().split(" ")
nextCharArray = lambda: list(next())
nextIntArray = lambda: list(map(int, nextStrArray()))
nextLongArray = lambda: list(map(int, nextStrArray()))
myconv = lambda list, no: " ".join(list) if no == 8 else "\n".join(list) if no == 9 else "".join(list)
solve = lambda: (out(f"{sum(1 for i in range(N) if max <= list[i])}\n"), flush())

CONVSTR = {8: " ", 9: "\n", 0: ""}
N = nextInt()
list = nextIntArray()
max = 0
output = 0
for i in range(N):
    if max <= list[i]:
        max = list[i]
        output += 1
myout(output) # End of Code
